cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Queries.hs

```diff
@@ -78,56 +80,101 @@ hprop_cli_queries = H.integrationWorkspace "cli-queries" $ \tempAbsBasePath' ->
-  -- stake-pools to a file
-  let stakePoolsOutFile = work </> "stake-pools-out.json"
-  H.noteM_ $ H.execCli' execConfig [ "conway", "query", "stake-pools"
-                                   , "--out-file", stakePoolsOutFile]
-
-  -- query drep-state to stdout
-  -- TODO: deserialize to a Haskell value when
-  -- https://github.com/IntersectMBO/cardano-cli/issues/606 is tackled
-  dreps :: Aeson.Value <- H.noteShowM $ H.execCliStdoutToJson execConfig [ "conway", "query", "drep-state", "--all-dreps"]
-  assertArrayOfSize dreps 3
-  -- query drep-state to a file
-  let drepStateOutFile = work </> "drep-state-out.json"
-  H.noteM_ $ H.execCli' execConfig [ "conway", "query", "drep-state", "--all-dreps"
-                                   , "--out-file", drepStateOutFile]
-  _ :: Aeson.Value <- H.leftFailM . H.readJsonFile $ drepStateOutFile
+
+  -- protocol-parameters
+  do
+    -- to stdout
+    protocolParametersOut <- H.execCli' execConfig [ eraName, "query", "protocol-parameters" ]
+    H.diffVsGoldenFile
+      protocolParametersOut
+      "test/cardano-testnet-test/files/golden/queries/protocolParametersOut.txt"
+    -- protocol-parameters to a file
+    let protocolParametersOutFile = work </> "protocol-parameters-out.json"
+    H.noteM_ $ H.execCli' execConfig [ eraName, "query", "protocol-parameters"
+                                     , "--out-file", protocolParametersOutFile]
+    H.diffFileVsGoldenFile
+      protocolParametersOutFile
+      "test/cardano-testnet-test/files/golden/queries/protocolParametersFileOut.json"
+
+  -- tip
+  do
+    -- to stdout
+    _ :: QueryTipLocalStateOutput <- H.noteShowM $ H.execCliStdoutToJson execConfig [ eraName, "query", "tip" ]
+    -- to a file
+    let tipOutFile = work </> "tip-out.json"
+    H.noteM_ $ H.execCli' execConfig [ eraName, "query", "tip"
+                                     , "--out-file", tipOutFile]
+    _ :: QueryTipLocalStateOutput <- H.readJsonFileOk tipOutFile
+    pure ()
+
+  -- stake-pools
+  do
+    -- to stdout
+    stakePoolsOut <- H.execCli' execConfig [ eraName, "query", "stake-pools" ]
+    H.assertWith stakePoolsOut $ \pools ->
+      length (lines pools) == 3 -- Because, by default, 3 stake pools are created
+    -- Light test of the query's answer, the ids should exist:
+    forM_ (lines stakePoolsOut) $ \stakePoolId -> do
+      H.execCli' execConfig [ eraName, "query", "pool-state"
+                            , "--stake-pool-id", stakePoolId ]
+    -- to a file
+    let stakePoolsOutFile = work </> "stake-pools-out.json"
+    H.noteM_ $ H.execCli' execConfig [ eraName, "query", "stake-pools" , "--out-file", stakePoolsOutFile]
+
+  -- stake-distribution
+  do
+    -- to stdout
+    stakeDistrOut <- H.execCli' execConfig [ eraName, "query", "stake-distribution" ]
+    -- stake addresses with stake
+    let stakeAddresses = map (both T.strip . T.breakOn " " . T.strip . fromString) . drop 2 . lines $ stakeDistrOut
```

Comment:
Does this not return JSON?

Reply:
when writing to a file - yes. When writing to stdout, nope, plain text.

Reply:
That's not good, I'll fix that.